### ISSUE TITLE
Build fails with missing "alexandria" dependency

### ISSUE THREAD

**Issue by MithicSpirit** [atlas-engineer/nyxt#1816]
**Describe the bug**
When building with `make all` (no other envvars set), the build errors out stating 
```lisp
Unhandled ASDF/FIND-COMPONENT:MISSING-DEPENDENCY in thread #<SB-THREAD:THREAD "main thread" RUNNING
                                                              {1004DE0273}>:
  Component ASDF/USER::ALEXANDRIA not found, required by #<SYSTEM "nyxt">
```
(which is followed by a backtrace).

**Precise recipe to reproduce the issue**
This was originally done on my regular Arch Linux install, though for easier reproducibility I used docker:
0. Install docker; start it with `sudo docker run -it --rm archlinux:base-devel`
1. Install dependencies for Arch Linux (select `2` for `sbcl` when prompted for a provider for `cl-asdf`, though this issue also occurs if `ecl` is selected):
```sh
sudo pacman -Syu git sbcl cl-asdf webkit2gtk glib-networking gsettings-desktop-schemas enchant
```
2. Clone `nyxt` and enter its directory:
```sh
git clone 'https://github.com/atlas-engineer/nyxt.git'
cd nyxt
```
3. Build nyxt with "`make all`"
4. Wait a while

**Information**
- OS name+version: Arch Linux (rolling)
- Graphics card and driver: Irrelevant
- Desktop environment / Window manager name+version: Main system is on `qtile` (on top of `xfce`), docker only on TTY though
- How you installed Nyxt (Guix pack, package manager, build from source): `make all`
- Information from `show-system-information` (from previous build on main system):
```
Nyxt version: 2.1.1-661-gb5960ffe
Renderer version: GI-GTK
Operating system kernel: Linux 5.14.11-xanmod1-1
Lisp implementation: SBCL 2.1.7 (Dynamic space size: 1073741824)
Features: (:WEBKIT2 :WEBKIT2-2.32 :WEBKIT2-PASTE-PLAINTEXT :WEBKIT2-TRACKING
 :WEBKIT2-MUTE :WEBKIT2-EMOJI :WEBKIT2-MEDIA :WEBKIT2-SANDBOXING :GTK-3-22
 :GTK-3-20 :GTK-3-18 :GTK-3-16 :GTK-3-14 :GTK-3-12 :GTK-3-10 :GTK-3-8 :GTK-3-6
 :GTK-3-4 :GTK :GDK-3-22 :GDK-3-20 :GDK-3-18 :GDK-3-16 :GDK-3-14 :GDK-3-12
 :GDK-3-10 :GDK-3-8 :GDK-3-6 :GDK-3-4 :CAIRO-1-10 :CAIRO-1-12 :GDK-PIXBUF
 :GLIB-2-30 :GLIB-2-32 :GLIB-2-34 :GLIB-2-36 :GLIB-2-38 :GLIB-2-40 :GLIB-2-42
 :GLIB-2-44 :GLIB-2-46 :GLIB-2-48 :GLIB-2-50 :GLIB-2-52 :GLIB-2-54 :GLIB-2-56
 :GLIB-2-58 :GLIB :NYXT-2 :FSET-EXT-STRINGS :CUSTOM-HASH-TABLE-NATIVE :SWANK
 :PLUMP-UTF-32 :CL-PPCRE-UNICODE :GLOBAL-VARS :DECLARE-TYPES :PARENSCRIPT
 :SBCL+SAFE-STANDARD-READTABLE :NAMED-READTABLES :LPARALLEL
 :LPARALLEL.WITH-CLTL2 :LPARALLEL.WITH-CAS :LPARALLEL.WITH-STEALING-SCHEDULER
 :21BIT-CHARS :CHUNGA CHIPZ-SYSTEM:GRAY-STREAMS :FAST-IO-SV :FAST-IO
 :SBCL-USES-SB-ROTATE-BYTE :ASDF-SYSTEM-CONNECTIONS :CLOSER-MOP :SPLIT-SEQUENCE
 :CL-UNICODE :FLEXI-STREAMS :CL-PPCRE :CL-JSON-DOUBLE-FLOAT-IS-SUBSUMED
 :CL-JSON-SINGLE-FLOAT-IS-SUBSUMED :CL-JSON-CLOS :CL-JSON :BORDEAUX-THREADS
 :THREAD-SUPPORT CFFI-FEATURES:FLAT-NAMESPACE CFFI-FEATURES:X86-64
 CFFI-FEATURES:UNIX :CFFI CFFI-SYS::FLAT-NAMESPACE ALEXANDRIA::SEQUENCE-EMPTYP
 :QUICKLISP :ASDF3.3 :ASDF3.2 :ASDF3.1 :ASDF3 :ASDF2 :ASDF :OS-UNIX
 :NON-BASE-CHARS-EXIST-P :ASDF-UNICODE :X86-64 :GENCGC :64-BIT :ANSI-CL
 :COMMON-LISP :ELF :IEEE-FLOATING-POINT :LINUX :LITTLE-ENDIAN
 :PACKAGE-LOCAL-NICKNAMES :SB-CORE-COMPRESSION :SB-LDB :SB-PACKAGE-LOCKS
 :SB-THREAD :SB-UNICODE :SBCL :UNIX)

ASDF version: 3.3.1
Critical dependencies: (/build/nyxt-browser-git/src/nyxt-browser/_build/submodules/cl-cffi-gtk/gtk/cl-cffi-gtk.asd
 /build/nyxt-browser-git/src/nyxt-browser/_build/quicklisp-client/dists/quicklisp/software/cl-gobject-introspection-20210124-git/cl-gobject-introspection.asd
 /build/nyxt-browser-git/src/nyxt-browser/_build/submodules/cl-webkit/webkit2/cl-webkit2.asd)

Quicklisp dist version: 2021-08-07
Quicklisp client version: 2020-01-04
Local project directories: (/build/nyxt-browser-git/src/nyxt-browser/_build/submodules/
 /build/nyxt-browser-git/src/nyxt-browser/_build/quicklisp-client/local-projects/)
Critical dependencies(#<SYSTEM cl-cffi-gtk / cl-cffi-gtk-20201220-git / quicklisp 2021-08-07>
 #<SYSTEM cl-gobject-introspection / cl-gobject-introspection-20210124-git / quicklisp 2021-08-07>
 #<SYSTEM cl-webkit2 / cl-webkit-20210807-git / quicklisp 2021-08-07>)
```

**Comment by jmercouris** [comment]
There seems to be some odd recursion going on for your local project directories.

**Comment by MithicSpirit** [comment]
I can also reproduce this on the `ubuntu` docker image, so the issue almost certainly is not with my system (unless it somehow also affects docker?)

**Comment by jmercouris** [comment]
I see, thank you for the information, I'm investigating right now. If I can't come up with something, hopefully @Ambrevar has some ideas.

P.S. other individuals on our IRC channel have reported the same issue.

**Comment by jmercouris** [comment]
I confirm the fix. Thank you Pierre! Please feel free to reopen if you are experiencing any issues!

**Comment by Vorlonhomeworld** [comment]
I'm having the same issue as well. I'm using Garuda Linux with SBCL. I can call Alexandria from SBCL directly, however, according to nyxt I don't have that software installed.  My exact error message is:

Component ASDF/USER::ALEXANDRIA not found, required by #<NASDF-SYSTEM "nyxt">

**Comment by aadcg** [comment]
@Vorlonhomeworld As mentioned in the thread, ensure that the directory `_build ` is populated with git submodules.

Also see `make help `, which explains the effect of env var `NYXT_SUBMODULES`.

**Comment by Vorlonhomeworld** [comment]
OK, I downloaded it recursively from GitHub.  It's working now.

**Comment by simon-brooke** [comment]
Thanks, @Ambrevar, your fix worked for me (Debian Bookworm, SBCL 2.2.9.debian).
